### DISCUSSION TITLE
How to config keycloak for RabbitMQ

### DISCUSSION THREAD

**Discussion by harrybui2804** [rabbitmq/rabbitmq-server#11660] [Other] [answered]
### Describe the bug

I am configuring Keycloak for RabbitMQ. I have logged in with Keycloak to get the token. When I return to RabbitMQ, I get an error.
http://127.0.0.1:15763/api/whoami 401 (Unauthorized)
Error as shown below
<img width="972" alt="Screenshot 2024-07-10 at 16 39 07" src="https://github.com/rabbitmq/rabbitmq-server/assets/39423618/c98a5d4c-2200-4df9-ae3c-4fa3f62ecc77">

### Reproduction steps

My config file (rabbitmq.conf):
management.oauth_enabled = true
management.oauth_client_id = auth_mqtt
management.oauth_provider_url = http://localhost:8282/realms/mqtt
management.oauth_scopes = openid profile

auth_backends.1 = rabbit_auth_backend_oauth2
auth_oauth2.resource_server_id = rabbit@vm-win2k16
auth_oauth2.preferred_username_claims.1 = preferred_username
auth_oauth2.additional_scopes_key = extra_scope

log.file.level = debug

### Expected behavior

Please help me how to fix it. Thanks very much

**Comment by michaelklishin** [comment, marked as answer]
Besides the [OAuth 2 documentation guide](https://www.rabbitmq.com/docs/oauth2) there are two others with [OAuth 2 troubleshooting recommendations](https://www.rabbitmq.com/docs/troubleshooting-oauth2) and [examples for several IDPs](https://www.rabbitmq.com/docs/oauth2-examples), including [Keycloak](https://www.rabbitmq.com/docs/oauth2-examples-keycloak).

Please take it from here. Our team [will not troubleshoot OAuth 2 for non-paying users](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#exceptions-question-that-will-be-ignored).
